### PR TITLE
Fix typo in date format

### LINKED ISSUE #26500: Typo in date format
**Elasticsearch version** (`bin/elasticsearch --version`):
master
**Plugins installed**: []
N/A
**JVM version** (`java -version`):
N/A
**OS version** (`uname -a` if on a Unix-like system):
N/A
**Description of the problem including expected versus actual behavior**:
I believe that there is a typo in org.elasticsearch.common.joda.Joda. As seen below, ```Time``` should not be capitalized, but rather should be ```basic_t_time```.

https://github.com/elastic/elasticsearch/blob/8f0369296f61d8c6ddcf821899715b4d969d4438/core/src/main/java/org/elasticsearch/common/joda/Joda.java#L82

### PR DESCRIPTION
**Pull request by liketic**
Fix #26500

### CODE REVIEW

**Comment by jpountz** [comment]
@liketic Thanks for the PR. Would you mind adding a test to avoid regressions in the future?

File: core/src/test/java/org/elasticsearch/common/joda/JodaTests.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        assertIllegalFormatException("basic_T_time");
+    }
+
+    private void assertIllegalFormatException(String format) {
+        try {
+            Joda.forPattern(format);
+            fail("Pattern: " + format);
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage().contains("Invalid format: [" + format + "]"), equalTo(true));
+        }
```
Comment (jpountz) [inline core/src/test/java/org/elasticsearch/common/joda/JodaTests.java]: can you use `expectThrows` instead?

**Review by jpountz — APPROVED** [review]
Thanks, it looks good to me, I just left a minor comment about using expectThrows instead of try/fail/catch/assert for testing.

File: core/src/test/java/org/elasticsearch/common/joda/JodaTests.java (hunk 1 above)
Comment (liketic) [reply core/src/test/java/org/elasticsearch/common/joda/JodaTests.java]: Yes, thanks.

**Pull request merged**